timeservice.go

```diff
@@ -75,6 +75,28 @@ type ntpReferenceClockSCION struct {
 	pather     *scion.Pather
 }
 
+const tlsCertReloadInterval = time.Minute * 10
+
+type tlsCertCache struct {
+	cert     *tls.Certificate
+	updated  time.Time
+	certFile string
+	keyFile  string
+}
+
+func (c *tlsCertCache) loadCert(chi *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	tNow := time.Now()
+	if c.updated.Add(tlsCertReloadInterval).Before(tNow) {
```

Comment:
I would write this like
```
if now.Before(c.reloadedAt) || !now.Before(c.reloadedAt.Add(tlsCertReloadInterval)) {
```

Renamed variables, robuster against clock jumping backwards, slightly changed semantics at the end of the interval.